detectorCommon/src/xtalk_k4geo.cpp

```diff
@@ -0,0 +1,245 @@
+      // Do we need to check if the index neighbour theta is valid?
+      if (theta_id_neighbour >= theta_id && theta_id_neighbour < (theta_id + aSeg.mergedThetaCells(layer_id))) { // crosstalk neighbour type 1
+        dummy_xtalk = dummy_xtalk_radial;
+      }
+      else if (theta_id_neighbour < theta_id && theta_id_neighbour > (theta_id - aSeg.mergedThetaCells(layer_id+deltaLayer))) { // crosstalk neighbour type 1
+        dummy_xtalk = dummy_xtalk_radial;
+      }
+      else if (theta_id_neighbour == (theta_id + aSeg.mergedThetaCells(layer_id))) { // crosstalk neighbour type 3
+        dummy_xtalk = dummy_xtalk_diagonal;
+      }
+      else if (theta_id_neighbour == (theta_id - aSeg.mergedThetaCells(layer_id+deltaLayer))) { // crosstalk neighbour type 3
+        dummy_xtalk = dummy_xtalk_diagonal;
+      }
+      else continue; 
+      
+      // check if neighbour theta index is valid
+      if (theta_id_neighbour < aFieldExtremes_layer[layer_indice_offset + deltaLayer][idThetaField].first || theta_id_neighbour > aFieldExtremes_layer[layer_indice_offset + deltaLayer][idThetaField].second) continue; 
+      aDecoder.set(cID, aSeg.fieldNameTheta(), theta_id_neighbour);
+      xtalk_neighbours.emplace_back(std::make_pair(cID, dummy_xtalk));
+    }
+  }
+
+  // cross talk neighbour type 2
+  // for neighbours in module/theta direction at same layer_id, do +-nMergedCells instead of +-1
+  // reset cellID
+  aDecoder.set(cID, aSeg.fieldNameLayer(), layer_id);
+  aDecoder.set(cID, aSeg.fieldNameTheta(), theta_id);
+  // loop over theta cells
+  for (int i=-1; i<=1; i+=2) {
+    // calculate theta_id of neighbour
+    int theta_id_neighbour = theta_id + i*aSeg.mergedThetaCells(layer_id);
+    // check if it is within the theta range of the layer
+    if (
+	(theta_id_neighbour < aFieldExtremes_layer[layer_indice_offset][idThetaField].first) ||
+	(theta_id_neighbour > aFieldExtremes_layer[layer_indice_offset][idThetaField].second)
+	) continue;
+    // set theta_id of cell ID
+    aDecoder[aSeg.fieldNameTheta()].set(cID, theta_id + i*aSeg.mergedThetaCells(layer_id));
+    xtalk_neighbours.emplace_back(std::make_pair(cID, dummy_xtalk_theta));
+  }
+ 
+  // cross talk type 4
+  // reset cellID
+  aDecoder.set(cID, aSeg.fieldNameLayer(), layer_id);
+  aDecoder.set(cID, aSeg.fieldNameTheta(), theta_id);
+  for (int loopLayer = aFieldExtremes_layer[0][idLayerField].first; loopLayer<=aFieldExtremes_layer[0][idLayerField].second; loopLayer++) {
+    // cells in adjacent layers have been studied, skip
+    if ((loopLayer - layer_id) <= 1 && (loopLayer - layer_id) >= -1) continue;
+    aDecoder.set(cID, aSeg.fieldNameLayer(), loopLayer);
+    int loopLayer_indice_offset = loopLayer - aFieldExtremes_layer[0][idLayerField].first;
+    // find crosstalk neighbours in the same theta tower
+    for (int i=-1; i <= aSeg.mergedThetaCells(layer_id); i++) {
+      int theta_id_neighbour = (theta_id + i) - ((theta_id + i) % aSeg.mergedThetaCells(loopLayer));
```

Comment:
Can `aSeg.mergedThetaCells(loopLayer)` be zero?

Reply:
Even though "aSeg.mergedThetaCells(loopLayer)" can never be 0 in the detector geometry, I have changed it to "(aSeg.mergedThetaCells(loopLayer) ? aSeg.mergedThetaCells(loopLayer):1)" just in case. Same for l122
https://github.com/zwu0922/k4geo/blob/main/detectorCommon/src/xtalk_neighbors_moduleThetaMergedSegmentation.cpp#L122